met-web/src/components/userManagement/userDetails/AddToEngagement.tsx

```diff
@@ -0,0 +1,229 @@
+        }
+    }, [JSON.stringify(formValues)]);
+
+    const { engagement: engagementErrors } = errors;
+
+    const handleClose = () => {
+        setAddUserModalOpen(false);
+        reset({});
+        setBackendError('');
+    };
+
+    const loadEngagements = async (searchText: string) => {
+        if (searchText.length < 3) {
+            return;
+        }
+        try {
+            setEngagementsLoading(true);
+            const response = await getEngagements({
+                search_text: searchText,
+            });
+            setEngagements(response.items);
+            setEngagementsLoading(false);
+        } catch (error) {
+            dispatch(
+                openNotification({
+                    severity: 'error',
+                    text: 'Error occurred while trying to fetch engagements, please refresh the page or try again at a later time',
+                }),
+            );
+            setEngagementsLoading(false);
+        }
+    };
+
+    const debounceLoadEngagements = useRef(
+        debounce((searchText: string) => {
+            loadEngagements(searchText).catch((error) => {
+                console.error('Error in debounceLoadEngagements:', error);
+            });
+        }, 1000),
+    ).current;
+
+    const addUserToEngagement = async (data: AddUserForm) => {
+        await addTeamMemberToEngagement({
+            user_id: savedUser?.external_id,
+            engagement_id: data.engagement?.id,
+        });
+        dispatch(
+            openNotification({
+                severity: 'success',
+                text: `You have successfully added ${savedUser?.username} as a Team Member on ${data.engagement?.name}.`,
+            }),
+        );
+    };
+
+    const setErrors = (error: AxiosError) => {
+        if (error.response?.status !== 409) {
+            return;
+        }
+        setBackendError(error.response?.data.message || '');
+    };
+
+    const onSubmit: SubmitHandler<AddUserForm> = async (data: AddUserForm) => {
+        try {
+            setIsAssigningRole(true);
+            await addUserToEngagement(data);
+            setIsAssigningRole(false);
+            handleClose();
+        } catch (error) {
+            setIsAssigningRole(false);
+            if (axios.isAxiosError(error)) {
+                setErrors(error);
+            }
+            dispatch(openNotification({ severity: 'error', text: 'An error occurred while trying to add user' }));
+        }
+    };
+
+    return (
+        <Modal open={addUserModalOpen} onClose={handleClose} keepMounted={false}>
+            <Paper sx={{ ...modalStyle }}>
+                <FormProvider {...methods}>
+                    <form onSubmit={handleSubmit(onSubmit)}>
+                        <Grid container direction="row" alignItems="flex-start" justifyContent="flex-start" spacing={2}>
+                            <Grid item xs={12}>
+                                <MetHeader3 bold>Add {savedUser?.username} to Engagement</MetHeader3>
```

Comment:
I think this should be first name and last name as in the user management page instead of username